File: 2022/4/part1.pl
#!/usr/bin/perl
use warnings;
use strict;

my $filename = '4/input.txt';

open(FH, '<', $filename) or die $!;

my $total = 0; 
while(my $line =<FH>){
    chomp $line;
    my @elves = split(",", $line);
    my @elve1 = split("-",$elves[0]);
    my @elve2 = split("-",$elves[1]);
    if(($elve1[0]<=$elve2[0] && $elve1[1]>=$elve2[1] )|| ($elve2[0]<=$elve1[0] && $elve2[1]>=$elve1[1])){
        $total = $total + 1;
    }
}

print($total);

close(FH);

File: 2023/1/part1.py
#!/usr/bin/python
totalsum = 0

with open('2023/1/input.txt') as f:
    lines = f.readlines()

for line in lines:
    line = line.rstrip('\n')
    number=""
    for ch in line:
        if ch in "0123456789":
            number += ch
            break
    
    for ch in line[::-1]:
        if ch in "0123456789":
            number += ch
            break
    
    print(number)
    totalsum += int(number)

print(f'Total sum is {totalsum}')
        

